### PR TITLE
Manual Engine roll for flutter/engine#7791 - Add trailing whitespace tracking.

### PR DESCRIPTION
**Pull request by GaryQian**
This fixes tests for https://github.com/flutter/engine/pull/7791 and rolls the engine.

Will include actual engine roll once engine-patch lands.

### CODE REVIEW

File: packages/flutter/test/painting/text_painter_test.dart [hunk 1]
```diff
@@ -133,7 +133,7 @@ void main() {
     Offset caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: 0), ui.Rect.zero);
     expect(caretOffset.dx, 21);
     caretOffset = painter.getOffsetForCaret(const ui.TextPosition(offset: text.length), ui.Rect.zero);
-    expect(caretOffset.dx, 399);
+    expect(caretOffset.dx, 441.0);
```
Comment (GaryQian) [inline packages/flutter/test/painting/text_painter_test.dart]: There are now metrics to track trailing whitespaces

**Comment by xster** [comment]
I'm somewhat surprised that that was the only thing needed fixing but LGTM

**Comment by GaryQian** [comment]
This only applies to centered/right aligned text with trailing whitespace. Not too many framework tests tested for this specific behavior (although it is tested well in the engine)

**Comment by dnfield** [comment]
Lgtm

**Comment by dnfield** [comment]
LGTM

**Comment by GaryQian** [comment]
Will add a framework test to make sure the behavior is fully translating to the caret positioning.

**Pull request merged**